packages/server/modules/core/repositories/users.ts

```diff
@@ -51,16 +51,30 @@ export async function getUsers(
   userIds = isArray(userIds) ? userIds : [userIds]
 
   const q = Users.knex<UserWithOptionalRole[]>().whereIn(Users.col.id, userIds)
+  q.leftJoin(
+    USER_EMAILS_TABLE_NAME,
+    `${USER_EMAILS_TABLE_NAME}.userId`,
+    Users.col.id
+  ).where({ primary: true })
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { email, verified, ...usersCols } = Users.col
+  const columns: (Knex.Raw<UserRecord> | string)[] = [
+    ...Object.values(usersCols),
+    `${USER_EMAILS_TABLE_NAME}.email`,
+    `${USER_EMAILS_TABLE_NAME}.verified`
+  ]
   if (withRole) {
-    q.columns([
-      ...Object.values(Users.col),
-      // Getting first role from grouped results
-      knex.raw(`(array_agg("server_acl"."role"))[1] as role`)
-    ])
-    q.leftJoin(ServerAcl.name, ServerAcl.col.userId, Users.col.id)
-    q.groupBy(Users.col.id)
+    // Getting first role from grouped results
+    columns.push(knex.raw(`(array_agg("server_acl"."role"))[1] as role`))
   }
 
+  q.columns(columns)
```

Comment:
1. ServerAcl only needs to be joined if `withRole`
2. Why group on email and verified too?

Reply:
1. I didn't change that logic; do you think this should be changed?
2. email and verified are grouped because it is a one-to-many relationship. Giving the `primary=true` condition makes it a unique connection but I still get an error running it.